packages/store/ts/config/v2/table.ts

```diff
@@ -37,28 +37,25 @@ export function isValidPrimaryKey<schema extends SchemaInput, scope extends Scop
   );
 }
 
-/** @deprecated */
-export function isTableInput(input: unknown): input is TableInput {
-  return (
-    typeof input === "object" &&
-    input !== null &&
-    hasOwnKey(input, "schema") &&
-    hasOwnKey(input, "key") &&
-    Array.isArray(input["key"])
-  );
-}
-
 export type validateKeys<validKeys extends PropertyKey, keys> = {
   [i in keyof keys]: keys[i] extends validKeys ? keys[i] : validKeys;
 };
 
-export type validateTable<input, scope extends Scope = AbiTypeScope> = {
+export type ValidateTableOptions = { inStoreContext: boolean };
+
+export type validateTable<
+  input,
+  scope extends Scope = AbiTypeScope,
+  options extends ValidateTableOptions = { inStoreContext: false },
```

Comment:
😬 don't love this but it's prob a big refactor otherwise?

Reply:
ooc why wouldn't this be on scope?

Reply:
yeah could be on scope!